### ISSUE TITLE
Analysing without covariates

### ISSUE THREAD

**Issue by ehsankhajouei** [jiabowang/GAPIT3#23]
Dear All, 

I have a favour to ask: I am new to GAPIT, and association studies in general (I'm trying to find my way around it as the start of my PhD study in plant genomics). Based on the information that I gathered from GAPIT manual and online sources, I came up with a basic workflow as follows: 

1) importing some libraries and sources from GAPIT repository
2) input files consist of a Phenotypic file and a Hapmap file
3) leaving GAPIT to decide about "kinship" and "covariates"

**# First, importing libraries and sources**
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(version = "3.13")
BiocManager::install('multtest', force = TRUE)

library('MASS') # required for ginv
library(multtest)
library(gplots)
library(compiler) #required for cmpfun
 @library(scatterplot3d)

source("http://zzlab.net/GAPIT/emma.txt")
source("http://zzlab.net/GAPIT/gapit_functions.txt")

**# Importing Phenotypic and Hapmap files**
myY = ...
myG = ...

**# Calling GAPIT function**
myGAPIT <- GAPIT(
  Y=myY[,c(1, "some_trait")],
  G=myG,
  model = c("GLM", "MLM", "MLMM", "Blink"),
  Multiple_analysis = TRUE)

I received results and found associated  SNPs. My main **CONCERN** is that I didn't define "covariates". Can you please have a look at the above workflow and tell me if this looks Okay or do I need to define covariates/PCAs? If YES, would you point me to the best methods to define covariates in polyploid plants, as I understood there are a number of available approaches. 
Thanks very much in advance. 

Best, 
Ehsan

**Comment by jiabowang** [comment]
Hi Ehsan,
Thanks for your using GAPIT and sharing your problem with me.

The covariation in the GAWS model is important to control population stratification. Usually, the PCs are used as covariation. But the number of PCs that should be used is unsettled. 

In your case,  there are no covariations in any model.

Sincerely,
Jiabo

**Comment by ehsankhajouei** [comment]
Hi Jiabo, 

Thanks very much for the reply. Well, I decided to leave PCAs/covariates empty so that GAPIT decides about them as it's written in the version 3 manual:

There are size types of input data: phenotype (Y), genotype in hapmap format (G), genotype data in numerical format (GD), genotype map (GM), kinship (K), and covariate variables (CV), see Table 2.1.Phenotypic data must be provided and the rest are optional, including genotype data, map, kinship, and covariate. Kinship can be provided by users or be generated from genotype data, or even omitted by suing BLINK method. Genotypic data may not be needed for genomic prediction if the kinship matrix is provided by the user. **_Covariate variables (fixed effects), such as population structure represented by the Q matrix (subpopulation proportion) or principal components (PCs), are optional._** GAPIT provides the option to calculate PCs from the genotypic data. All input files should be saved as a “Tab” delimited text file.

So, I analyzed Wheat data without defining PCAs/covariates and already got results. Would you please tell me what do you recommend for defining covariates in Wheat data? Do you recommend to let GAPIT decide about population stratification or do you recommend me to define covariates first and then run the GAPIT? Many thanks in advance. 

Best/Ehsan

**Comment by knausb** [comment]
Hi @ehsankhajouei , GAPIT does not appear to determine whether or not your data contains population structure. It provides the tools to analyze your data with or without population structure, but that appears to be a decision you need to make. It you decide population structure is important you can use software such as [Structure](https://web.stanford.edu/group/pritchardlab/structure.html), [fastStructure](http://rajanil.github.io/fastStructure/), or [ADMIXTURE](https://dalexander.github.io/admixture/) to create the input that GAPIT would look for. I have no idea if any of these options will work for polyploids so you will have to determine this yourself.

Good luck!
Brian

**Comment by ehsankhajouei** [comment]
Hi Brian and thanks for the reply and the recommendation. 

Yeah, today I found the chance to try it myself, so if I don't define PCAs/Covariates, GAPIT interprets it zero. Then, I would suggest the latest version of manual is quite misleading, as it says:

**Covariate variables (fixed effects), such as population structure represented by the Q matrix (subpopulation proportion) or principal components (PCs), are optional. GAPIT provides the option to calculate PCs from the genotypic data. All input files should be saved as a “Tab” delimited text file.**

In reality, if you don't define covariates, GAPIT considers zero. 

Best /Ehsan

**Comment by knausb** [comment]
Hi @ehsankhajouei , I'm not sure I follow you. My understanding is that if you do not specify PCAs/Covariate, then they are not used. This means they do not exist as part of the analysis. If they do not exist as part of the analysis then they should not contribute to it. I doubt that setting this to zero will result in a similar result. For example, an uninformative covariates table may include 3 groups where each sample has a 1/3 probability of belonging to each group. Because each sample has an equal probability of assignment to each of the three groups it is "uninformative." But it's not zero. Can you propose a wording of the documentation that conveys the information you feel needs to be there? It may be included in the next version.

**Comment by ehsankhajouei** [comment]
Thanks Brian, @knausb, a lot for your reply. Well, this is from my experience and with my data which is from wheat. I tried two analyses to compare the outcomes using MLM and MLMM, with one defining PCA.total=0, and the other one I did not define PCAs at all. So, the outcomes of both analyses were identical, meaning that following statement form GAPIT manual is not correct:

**Covariate variables (fixed effects), such as population structure represented by the Q matrix (subpopulation proportion) or principal components (PCs), are optional. GAPIT provides the option to calculate PCs from the genotypic data. All input files should be saved as a “Tab” delimited text file.**

So if I may, I would reword as follows:

**Covariate variables (fixed effects), such as population structure represented by the Q matrix (subpopulation proportion) or principal components (PCs), are optional. Note that the default value of PCs in GAPIT is zero. However, should you need to calculate population structure, GAPIT provides the option to calculate PCs from the genotypic data. All input files should be saved as a “Tab” delimited text file.**


Also, I have tried another modification in the function which didn't work for me. I wanted to use 'EMMA' instead of 'VanRaden' by calling 'kinship.algorithm' in the main function. However, in my trials, no matter what I defined, kinship was constructed based on 'VanRaden' method and I couldn't change the default. Basically, I received the same outcome by either:

**kinship.algorithm='VanRaden'** 

or:

**kinship.algorithm='EMMA'**

and GAPIT would show the below message regardless:

**"Calculating kinship with VanRaden method..."**

Can you please help me to sort this out?

**Comment by ehsankhajouei** [comment]
Thanks very much Brian (@knausb) for the perfect explanation and sorry for getting back to you with delay. Everything is clear now. 
And yes, I will start a new issue for that matter if I cannot figure it out myself. 

Best, 
Ehsan
